src/font-inliner.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * @fileOverview Import bitmap data into Scratch 3.0, resizing image as necessary.
+ */
+import {FONTS} from 'scratch-render-fonts';
+
+/**
+ * Helper to create an SVG element with the correct NS.
+ * @param {string} tagName Tag name for the element.
+ * @return {!DOMElement} Element created.
+ */
+const createSVGElement = function (tagName) {
+    return document.createElementNS(
+        'http://www.w3.org/2000/svg', tagName
+    );
+};
+
+/**
+ * Given SVG data, inline the fonts. This allows them to be rendered correctly when set
+ * as the source of an HTMLImageElement. Here is a note from tmickel:
+ *   // Inject fonts that are needed.
+ *   // It would be nice if there were another way to get the SVG-in-canvas
+ *   // to render the correct font family, but I couldn't find any other way.
+ *   // Other things I tried:
+ *   // Just injecting the font-family into the document: no effect.
+ *   // External stylesheet linked to by SVG: no effect.
+ *   // Using a <link> or <style>@import</style> to link to font-family
+ *   // injected into the document: no effect.
+ * @param {SVGElement} svgTag The SVG dom object
+ * @return {void}
+ */
+const inlineSvgFonts = function (svgTag) {
+    // Collect fonts that need injection.
+    const fontsNeeded = {};
+    const collectFonts = function collectFonts (domElement) {
+        if (domElement.getAttribute && domElement.getAttribute('font-family')) {
+            fontsNeeded[domElement.getAttribute('font-family')] = true;
```

Comment:
Does it make sense to use a Set instead?

Reply:
Now that you mention it, I'm not sure why transformText uses an object for fontsNeeded, and not say an array. Maybe it's kind of like having a typed array of strings?

I think I'll leave it as it was, unless you feel it's particularly confusing?

Reply:
I think a `Set` would make sense here -- in fact, I think the original code is using an object as a string-to-Boolean map / dictionary, which they're using to emulate a `Set`. I don't think it's critical to change it right this moment, but it would express the intent better and simplify the loop below as @kchadha observed.

Reply:
Done